src/js/helpers.js

```diff
@@ -63,3 +63,57 @@ export const handleClick = ( {
 		}
 	);
 };
+
+/**
+ * Make a request to a browser AI to generate text.
+ *
+ * @param {string} provider Provider to use.
+ * @param {string} prompt   Prompt to send to the API.
+ * @param {string} content  Content to add in addition to the prompt.
+ */
+export const browserAITextGeneration = async (
+	provider = '',
+	prompt = '',
+	content = ''
+) => {
+	switch ( provider ) {
+		case 'chrome_ai':
+			return chromeAITextGeneration( prompt, content );
+		default:
+			return '';
+	}
+};
+
+/**
+ * Make a request to the Chrome AI API to generate text.
+ *
+ * @param {string} prompt  Prompt to send to the API.
+ * @param {string} content Content to add in addition to the prompt.
+ */
+export const chromeAITextGeneration = async ( prompt = '', content = '' ) => {
+	let result = '';
+
+	if ( ! window.ai ) {
+		return result;
+	}
+
+	const supportsTextGeneration =
+		await window.ai.languageModel?.capabilities();
+
+	if (
+		supportsTextGeneration &&
+		supportsTextGeneration.available === 'readily'
+	) {
+		const session = await window.ai.languageModel.create( {
+			initialPrompts: [
+				{
+					role: 'system',
+					content: prompt,
+				},
+			],
+		} );
+		result = await session.prompt( `"""${ content }"""` );
```

Comment:
I came across some random errors while testing scenarios. We should consider adding error handling here to inform the user about failures. Additionally, we need to set a limit on the length of the content being sent in the prompt (the limit on my browser model appears to be 6,144 tokens).

![image](https://github.com/user-attachments/assets/1e56a4d8-c8d8-4fbd-83ef-b5f5fdfeabbc)

Reply:
I've added a token limit now to the Provider, so we trim the content we use if we are above the limit. I've also added some additional error handling, just outputting an `alert` if something goes wrong. Not the best UX but for this Chrome AI that's still under development, I think that's good enough for now.

Reply:
Yes, good enough for now.